test-cases/final-dump-test.sh

```
@@ -23,18 +28,18 @@ do
             -e 's/\[[0-9][0-9]* x i8\]/[?? x i8]/g' \
         > $out
 	# Add a newline to the end of a file if there isn't to resolve platform differences.
-	ed -s $out <<< w > /dev/null 2>&1
-	if [ ! -r $exp ] ; then 
+	ed -s $out <<< w > /dev/null 2>&1 || abort "failed to add newline to output, is ed installed?"
```

I realised that this is why the tests were failing locally last week - I didn't have `ed` installed so it silently failed to add newlines.

I added this as wasn't sure if someone in the future might encounter the same issue? Alternatively I could add a note in [INSTALL.md](https://github.com/pschachte/wybe/blob/master/INSTALL.md) rather than modifying this script?

---

What good is a Unix environment if there's no `ed` installed :(

Probably a good thing to note in INSTALL, but only as something required for tests--something a user wouldn't really be doing

---

Scouring SO I have found [this](https://unix.stackexchange.com/a/263965):
```
[ -n "$(tail -c1 file)" ] && echo >> file
```
The poster claims this is POSIX compliant, and also works without `ed` :)